modules/types/src/network.ts

```diff
@@ -1,9 +1,253 @@
+*  @ethersproject/providers/FallbackProvider in order to fallback to other providers in the
+*  event of failed requests.
+*/
+export class ChainRpcProvider implements Provider {
+  readonly chainId: number;
+  readonly providerUrls: string[];
+  readonly _provider: JsonRpcProvider | FallbackProvider;
+
+  RPC_TIMEOUT: number = 10_000;
+  _isProvider: boolean = true;
+  _networkPromise: Promise<Network>;
+  _network: Network;
+  anyNetwork: boolean = false;
+
+  constructor(chainId: number, providers: string[] | JsonRpcProvider[], stallTimeout?: number) {
+    // We'll collect all the provider URLs as we hydrate each provider.
+    var providerUrls: string[] = [];
+    var provider: JsonRpcProvider | FallbackProvider;
+    if (providers.length > 1) {
+      provider = new FallbackProvider(
+        // Map the provider URLs into JsonRpcProviders 
+        providers.map((provider: string | JsonRpcProvider, priority: number) => {
+          const hydratedProvider = (typeof(provider) === "string") ? new JsonRpcProvider(provider, chainId) : provider;
+          providerUrls.push(hydratedProvider.connection.url);
+          return {
+            provider: hydratedProvider,
+            priority: priority,
+            // Timeout before also triggering the next provider; this does not stop
+            // this provider and if its result comes back before a quorum is reached
+            // it will be incorporated into the vote
+            // - lower values will cause more network traffic but may result in a
+            //   faster retult.
+            // TODO: Should we have our own default timeout defined, as well as a config option for this?
+            // Default timeout is written as either 2sec or .75sec (in @ethers-project/fallback-provider.ts):
+            // config.stallTimeout = isCommunityResource(configOrProvider) ? 2000: 750;
+            stallTimeout,
+            weight: 1
+          }
+        }),
+        // Quorum stays at 1, since we only ever want to send reqs to 1 node at a time.
+        1
+      );
+    } else if (providers.length === 1) {
+      const singleProvider = providers[0];
+      provider = (typeof(singleProvider) === "string") ? new JsonRpcProvider(singleProvider, chainId) : singleProvider;
+    } else {
+      throw new Error("At least one provider must be defined.")
+    }
+
+    this._networkPromise = provider.getNetwork();
+    this._network = provider.network;
+
+    this._provider = provider;
+    this.chainId = chainId;
+    this.providerUrls = providerUrls;
+  }
+
+  send(method: string, params: any[]): Promise<any> {
+    if (this._provider instanceof JsonRpcProvider) {
+      return (this._provider as JsonRpcProvider).send(method, params);
+    } else {
+      const providers = (this._provider as FallbackProvider).providerConfigs.map(p => p.provider as JsonRpcProvider);
+      var errors: Error[] = [];
+      return Promise.race<any>(
```

Comment:
Note that here is where we are executing the `Promise.race`. Essentially, in any other method call in this class, we literally just call the underlying provider. Fallback provider does this sort of 'race-like' operation we see here internally, under the hood.

However, there is no `FallbackProvider.send` method, so I've taken the liberty of implementing this myself. Assuming we had multiple provider URLs passed into constructor (and therefore we're using `FallbackProvider`), we're basically executing the RPC `send` call on every child provider (AS `JsonRpcProvider`) belonging to our `FallbackProvider`, and using a Promise.race to return the fastest one.

A delay for timeout of 10s rejects the Promise. Additionally, if all `send` calls error out, we reject the Promise with a list of errors.

Reply:
Going to add the above in descriptive comments here.